extensions/replication/ReplicationAPI.js

```diff
@@ -0,0 +1,100 @@
+     * request (for metrics accounting)
+     * @param {string} params.fromLocation - source location name (for
+     * metrics accounting)
+     * @param {string} params.contentLength - content length (for
+     * metrics accounting)
+     * @param {string} [params.resultsTopic] - name of topic to get
+     * result message (no result will be published if not provided)
+     * @return {ActionQueueEntry} new action entry
+     */
+    static createCopyLocationAction(params) {
+        const action = ActionQueueEntry.create('copyLocation');
+        action
+            .setAttribute('target', {
+                bucket: params.bucketName,
+                key: params.objectKey,
+                version: params.versionId,
+                eTag: params.eTag,
+                lastModified: params.lastModified,
+            })
+            .setAttribute('toLocation', params.toLocation)
+            .setAttribute('metrics', {
+                origin: params.originLabel,
+                fromLocation: params.fromLocation,
+                contentLength: params.contentLength,
+            });
+        if (params.resultsTopic) {
+            action.setResultsTopic(params.resultsTopic);
+        }
+        return action;
+    }
+
+    /**
+     * Send an action to the data mover service
+     *
+     * @param {BackbeatProducer} producer - backbeat producer instance
+     * @param {ActionQueueEntry} action - The action entry to send to
+     * the data mover service
+     * @param {Logger.newRequestLogger} log - logger object
+     * @param {Function} cb - callback: cb(err)
+     * @return {undefined}
+     */
+    static sendDataMoverAction(producer, action, log, cb) {
+        const { bucket, key } = action.getAttribute('target');
+        const kafkaEntries = [{ key: `${bucket}/${key}`,
+                                message: action.toKafkaMessage() }];
+        producer.sendToTopic(dataMoverTopic, kafkaEntries, (err, reports) => {
+            if (err) {
+                log.error('could not send data mover action',
+                          Object.assign({
+                              method: 'ReplicationAPI.sendDataMoverAction',
+                              error: err,
+                          }, action.getLogInfo()));
+                return cb(err);
+            }
+            log.debug('sent action to the data mover',
+                      Object.assign({
+                          method: 'ReplicationAPI.sendDataMoverAction',
+                      }, action.getLogInfo()));
+            const { origin, fromLocation, contentLength } =
+                  action.getAttribute('metrics');
+            ReplicationMetrics.onReplicationQueued(
+                origin, fromLocation, action.getAttribute('toLocation'),
+                contentLength, reports[0].partition);
+            return cb();
+        });
+    }
+
+    static getDataMoverTopic() {
+        return dataMoverTopic;
+    }
+
+    static setDataMoverTopic(newTopicName) {
```

Comment:
maybe being too picky, but I was thinking setters shouldn't be static methods because there is state that is changing

Reply:
Actually I added this setter/getter for lifecycle unit tests only as they set a specific topic name instead of the configured one. I think it's ok as it avoids to have to create a ReplicationAPI class instance which otherwise would not be of much use.